Assets/Scripts/UI/AchievementPatches.cs

```diff
@@ -0,0 +1,121 @@
+using Managers;
+using Save;
+using UI;
+using UnityEngine.UI;
+using UnityEngine;
+
+public class AchievementPatches : MonoBehaviour
+{
+    [SerializeField] private Image timeImage,candyImage,deathImage, playPanel;
+    [SerializeField] private Sprite defaultTime, defaultDeath, timePatch, candyPatch, deathPatch, goldPanel, defaultPanel;
+    private Sprite defaultCandy; //not serialized because passed in as a param.
+    [SerializeField, Tooltip("Time to bead to gain achievementPatch in seconds")] 
+    private float level1Threshold, level2Threshold, level3Threshold, level4Threshold;
+
+    /// <summary>
+    /// A check for whether the panel should be gold for each level
+    /// </summary>
+    public bool isGoldL1 { get; private set; } = false;
+    public bool isGoldL2 { get; private set; } = false;
+    public bool isGoldL3 { get; private set; } = false;
+    public bool isGoldL4 { get; private set; } = false;
+
+    /// <summary>
+    /// Displays achievement patches if applicable, called by Level Select line 97
+    /// </summary>
+    /// <param name="level">Level number</param>
+    /// <param name="timeTaken">Time taken to complete level in seconds</param>
+    /// <param name="candiesCollected">Highest number of candies collected in level</param>
+    /// <param name="candiesInLevel">Total number of candies in level</param>
+    /// <param name="numDeaths">Lowest number of deaths in level</param>
+    /// <param name="levelCandy">The candy sprite for level</param>
+    public void DisplayAchievementPatches(int level, float timeTaken, int candiesCollected, int candiesInLevel, int 
+            numDeaths, Sprite levelCandy)
+    {
+        //conditions to meet to get patch
+        bool candyCondition = candiesInLevel - candiesCollected == 0;
+        bool deathCondition = numDeaths == 0;
+        bool timeCondition = WithinTimeThreshold(level, timeTaken);
+        
+        //sets the candy image (because it changes depending on level)
+        defaultCandy = levelCandy;
+        
+        //check default value for candy
+        if (!candiesInLevel.Equals(-1))
```

Comment:
nit
```suggestion
        if (candiesInLevel != -1)
```

also candyCondition should also probably just check for -1